src/indexhm3.ts

```diff
@@ -0,0 +1,188 @@
+class School {
+    // implement 'add area', 'remove area', 'add lecturer', and 'remove lecturer' methods
+
+    private _areas: Area[] = [];
+    private _lecturers: object[] = []; // Name, surname, position, company, experience, courses, contacts
+
+    get areas(): Area[] {
+        return this._areas;
+    }
+
+    get lecturers(): object[] {
+        return this._lecturers;
+    }
+
+    addArea(area: Area): void {
+        this._areas.push(area);
+    }
+
+
+    removeArea(name: string): void {
+        const index = this._areas.findIndex((i: Area) => name === i.name);
+        if (index !== -1) {
+            this._areas.splice(index, 1);
+        }
+    }
+
+    addLecturer(lecturer: object): void {
+        this._lecturers.push(lecturer);
+    }
+
+    removeLecturer(name: string): void {
+        const index = this._lecturers.findIndex((i: any) => name === i.name);
+        if (index !== -1) {
+            this._lecturers.splice(index, 1);
+        }
+    }
+}
+
+class Area {
+    // implement getters for fields and 'add/remove level' methods
+    private _levels: Level[] = [];
+    _name: string;
+
+    constructor(name: string) {
+        this._name = name;
+    }
+
+    get name(): string {
+        return this._name;
+    }
+
+    get levels(): Level[] {
+        return this._levels;
+    }
+
+    addLevel(level: Level): void {
+        this._levels.push(level);
+    }
+
+
+    removeLevel(name: string): void {
+        const index = this._levels.findIndex((i: Level) => name === i.name);
+        if (index !== -1) {
+            this._levels.splice(index, 1);
+        }
+    }
+
+
+}
+
+class Level {
+    // implement getters for fields and 'add/remove group' methods
+
+    _groups: Group[] = [];
+    _description:string;
+    _name: string;
+
+    constructor(name: string, description: string) {
+        this._name = name;
+        this._description = description;
+    }
+
+    get name(): string {
+        return this._name;
+    }
+
+    get description(): string {
+        return this._description;
+    }
+
+    addGroup(group: Group): void {
+        this._groups.push(group);
+    }
+
+    removeGroup(directionName: string): void {
+        const index = this._groups.findIndex((i: Group) => directionName === i.directionName);
+        if (index !== -1) {
+            this._groups.splice(index, 1);
+        }
+    }
+
+}
+
+class Group {
+    // implement getters for fields and 'add/remove student' and 'set status' methods
```

Comment:
I'm sorry, I couldn't locate the getters for the fields. Could you please add them?

Reply:
And the setter accessor for status should be used instead of the addStatus method.